configuration_manager/directories_manager.py

```diff
@@ -0,0 +1,55 @@
+import os
+from utils.util import safe_makedir
+from datetime import datetime
+from definitions import OUTPUT_DIR
+
+
+class MetaDirectoriesManager(type):
+    """This class implements singleton for DirectoriesManager class."""
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(MetaDirectoriesManager, 
+                                                cls).__call__(*args, **kwargs)
+            cls.__output_directory = cls.__setup_output_directory(OUTPUT_DIR)
+            cls.get_results_directory()
+            cls.get_logs_directory()
+        return cls._instances[cls]
+
+    @staticmethod
+    def __setup_output_directory(name):
+        """Creates output directory for results, logs etc.
+
+        Parameters
+        ----------
+        name : str
+            Name for the Output directory
+
+        Returns
+        ------
+        target_directory: str
+            Output directory path
+        """
+        target_directory = os.path.join(OUTPUT_DIR, name)
+        target_directory += datetime.strftime(datetime.now(), 
+                                                    '_%Y-%m-%d_%H-%M')
+        safe_makedir(target_directory)
```

Comment:
Great idea. 
I only wonder if a conflict can still occur: If you start the software/cosim multiple times per MINUTE, would all results be in the same folder?
Maybe also add some kind of counter to the folder name?

Reply:
That's a good idea to uniquely identify the simultaneous runs by different users. Let's do it in next iterations.